### DISCUSSION TITLE
MET not compiling on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1636] [Configuration] [answered]
Good morning.

**Comment by georgemccabe** [comment]
Hi @whatheway, it looks posted this question, then edited the text to remove the actual question. Do you still need help with this issue?

**Reply by ghost** [reply]
No figured it out but I forgot to post the solution.

**Comment by georgemccabe** [comment]
Hi @whatheway, that is good to hear. Would you mind posting your solution so that other users that encounter a similar issue can refer to it? Thanks.

**Comment by ghost** [comment, marked as answer]
@georgemccabe 

For anyone using the community code located at the bottom of the METPlus download page https://dtcenter.org/community-code/metplus/download

**Other Installation Options

Will Hatheway, a meteorologist in Austin, TX, created BASH scripts to self install all the required files to run the WRF-4.3 and WRFCHEM-4.3.  Due to requests from the community, he added BASH scripts to install the MET and METplus.  These scripts are available on his [GitHub page](https://github.com/whatheway) and are supported by him.**

The code as an issue with python libraries being called on Ubuntu 22.04.  Python3 is called in the met install code using /usr/bin/python3 --version.  If that version is different then the python version called in code below it will break the code.  As the author of the code I am now aware of the issue and am trying to find a resolution to patch the code within the coming days.

`export MET_PYTHON_CC=-I${MET_PYTHON}/include/python3.8  
export MET_PYTHON_LD=-L${MET_PYTHON}/lib/python3.8/config-3.8-x86_64-linux-gnu\ -L${MET_PYTHON}/lib\ -lpython3.8\ -lcrypt\ -lpthread\ -ldl\ -lutil\ -lm   
export SET_D64BIT=FALSE  `

So any user having issues would need to change the 3.8 to whatever the current version of python3 is installed.

**Reply by jprestop** [reply]
@jprestop editing to add back the initial text that was removed:
Good morning,

Running into issues install MET and METplus on Ubuntu 22.04 Jammy Jellyfish. There has been a lot of compiler issues with this latest update for ubuntu. Ubuntu 22.04 comes with compiles gfortran/gcc/g++ version 11 by default. MET seems to be compatible with only gfortran/gcc/g++ version 9.

Utilizing the community code developed by Mr. Hatheway listed on the MET website: https://dtcenter.org/community-code/metplus/download

There seems to be a compiler error somewhere in their the library installation or the met installation.

I have tried forcing version 9 of gfortran/gcc/g++ for the installation by exporting the fortran variables to version 9 and that did not work either.

Attached is the library logs, make log, and compile_met_all output

Any suggestions would be helpful.
[bufrlib.make.log](https://github.com/dtcenter/METplus/files/8800135/bufrlib.make.log)

Regards,
Will
[compile_met_all.txt](https://github.com/dtcenter/METplus/files/8800132/compile_met_all.txt)
[make.log](https://github.com/dtcenter/METplus/files/8800133/make.log)
